src/main/java/software/amazon/awssdk/crt/CrtResource.java

```diff
@@ -42,12 +46,42 @@ public static int getAllocatedNativeResourceCount() {
     public CrtResource() {
     }
 
+    /**
+     * Marks a SubResource as owned by this Resource. This will cause the SubResource to be closed after this Resource
+     * is closed.
+     * @param subresource The owned subresource
+     * @return The original subresource.
+     */
+    public <T extends CrtResource> T own(T subresource) {
+        if (!isNull()) {
+            throw new IllegalStateException("Parent Resource already created and acquired, can't add sub-resources after the fact.");
+        }
+        ownedSubResources.push(subresource);
+        return subresource;
+    }
+
     protected void acquire(long _ptr) {
-        NATIVE_RESOURCES.put(_ptr, this.getClass().getCanonicalName());
+        if (!isNull()) {
+            throw new IllegalStateException("Can't acquire >1 Native Pointer");
+        }
+
+        if (_ptr == NULL) {
+            throw new IllegalStateException("Can't acquire NULL Pointer");
+        }
+
+        String lastValue = NATIVE_RESOURCES.put(_ptr, this.getClass().getCanonicalName());
+
+        if (lastValue != null) {
```

Comment:
This isn't a valid check, it is entirely possible to be given the same address when requesting memory for an object.

Reply:
Won't you run into use-after-free if you have two `CrtResource`'s with the same native pointer and you close one of them?

Reply:
No you're right, this is my mistake. Basically, there can only be 1 resource at a time per address. As long as dead resources are cleared (which they are), all is well. I'm going to chalk that one up to pre-coffee code review :)